sqlmesh/core/state_sync/engine_adapter.py

```diff
@@ -369,6 +394,67 @@ def _get_environment(
         env = self._environment_from_row(row)
         return env
 
+    @transactional()
+    def migrate(self) -> None:
+        super().migrate()
+
+    def _migrate_rows(self) -> None:
+        all_snapshots = self._get_snapshots(lock_for_update=True, validate_versions=False)
+        environments = self.get_environments()
+
+        snapshot_mapping = {}
+
+        for snapshot in all_snapshots.values():
+            models = {}
+            audits = {}
+
+            for snapshot_id in (snapshot.snapshot_id,) + snapshot.parents:
+                s = all_snapshots.get(snapshot_id)
+
+                if s:
+                    models[s.name] = s.model
+
+                    for audit in s.audits:
+                        audits[audit.name] = audit
+
+            new_snapshot = Snapshot.from_model(
+                snapshot.model,
+                physical_schema=snapshot.physical_schema,
+                models=models,
+                ttl=snapshot.ttl,
+                version=snapshot.version,
+                audits=audits,
+            )
+
+            if new_snapshot == snapshot:
```

Comment:
would it be cheaper to just compare fingerprints instead? Is there a reason to compare all of it?

Reply:
that's what eq does, it compares the fingerprints